pkg/pillar/cmd/volumemgr/volumemgr.go

```diff
@@ -455,6 +456,7 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject) in
 	if err := utils.WaitForVault(ps, log, agentName, warningTime, errorTime); err != nil {
 		log.Fatal(err)
 	}
+	ctx.vaultProcessed = true
```

Comment:
Couldn't we move the above GCInitialized loop and the WaitForVault before the subscriptions above so we do not end up called doUpdateVol() before we have the vault unlocked?